bin/internal/shared.bat

```diff
@@ -92,7 +92,13 @@ GOTO :after_subroutine
     SET update_dart_bin=%FLUTTER_ROOT%/bin/internal/update_dart_sdk.ps1
     REM Escape apostrophes from the executable path
     SET "update_dart_bin=!update_dart_bin:'=''!"
-    %powershell_executable% -ExecutionPolicy Bypass -Command "Unblock-File -Path '%update_dart_bin%'; & '%update_dart_bin%'"
+    REM PowerShell command must have exit code set in order to prevent all non-zero exit codes from being translated
+    REM into 1. The exit code 2 is used to detect the case where the major version is incorrect and there should be
+    REM no subsequent retries.
+    %powershell_executable% -ExecutionPolicy Bypass -Command "Unblock-File -Path '%update_dart_bin%'; & '%update_dart_bin%'; exit $LASTEXITCODE;"
+    IF "%ERRORLEVEL%" EQU "2" (
+      EXIT 1
```

Comment:
I _think_ this only exits you out of the subroutine and then actually not sure what it executes next (would it retry the subroutine again? That wouldn't make much sense since this error is unreoverable.). I would have expected this to set `exit_code` and then jump to `: final_exit`?

Reply:
/B is subroutine only - you can test this locally by removing the condition for the pwsh version detection, but it will immediately exit.

Reply:
I wonder why we do the 'final_exit' dance below then... But if you tried this and it works: LGTM